src/cmd/build.rs

```diff
@@ -33,6 +34,87 @@ use parity_wasm::elements::{External, MemoryType, Module, Section};
 /// This is the maximum number of pages available for a contract to allocate.
 const MAX_MEMORY_PAGES: u32 = 16;
 
+/// Describes which build artifacts to generate
+#[derive(Copy, Clone, Eq, PartialEq, Debug, StructOpt)]
+#[structopt(name = "build-artifacts")]
+pub enum BuildArtifacts {
+    /// Generate the Wasm, the metadata and a bundled `<name>.contract` file
+    #[structopt(name = "all")]
+    All,
+    /// Only the Wasm is created, generation of metadata and a bundled `<name>.contract` file is skipped
+    #[structopt(name = "code-only")]
+    CodeOnly,
+    /// Only the Wasm and the metadata are generated, no bundled `<name>.contract` file is created
```

Comment:
Yeah I thought about that too, it only works if there is an already existing wasm in the filesystem though. That's what you want to get at, right?

Reply:
It is not required, currently. It's just there as a nice to have. In the future the client tools could use that (if present) to verify the metadata matches the deployed code.

So I would say that if `metadata-only is specified then that field should be omitted, even if the wasm is already there because it is not guaranteed to match.

Reply:
When would you ever need `metadata-only` is my first question? Please tell me some use cases because I cannot come up with any that would not involve also building the `.wasm`.